frontend/src/components/posts/PostForm.jsx

```diff
@@ -1,42 +1,69 @@
 // NPM Packages
-import React, { useState } from 'react';
+import React, { useState } from "react";
+import PostsApi from "../../api/PostsApi";
 
 export default function PostForm({ onSubmit }) {
-  // Local State
-  const [contentText, setContentText] = useState('');
-  const postObject = require('../../assets/images/post.svg');
-  const postURL = postObject;
 
-  // Methods
-  const handleSubmit = () => {
-    // Invoke the passed in event callback
-    onSubmit({ contentText: contentText });
+    // Local State
+    const [contentText, setContentText] = useState("");
+    const postObject = require("../../assets/images/post.svg");
+    const postURL = postObject;
+    const [contentFile, setContentFile] = useState();
+    const [isFilePicked, setIsFilePicked] = useState(false);
+    const formData = new FormData();
 
-    // Clear the input field
-    setContentText('');
-  };
+    // Methods
+    const handleSubmit = (event) => {
+        if (isFilePicked) {
+            handleImagePostSubmit(event);
+        } else {
+            // Invoke the passed in event callback
+            onSubmit({ contentText: contentText });
+        }
+        // Clear the input field
+        setContentText("");
+    };
 
-  return (
-    <div className="postform-container">
-      <form className="postform">
-        <div>
-          <div>
-            <div className="postform-field">
-              <textarea 
-                className="postform-input"
-                placeholder="What's on your mind?"
-                value={contentText}
-                onChange={(e) => setContentText(e.target.value)}
-              />
-            </div>
+    const handleImagePostSubmit = (event) => {
+        event.preventDefault();
+        formData.append("file", contentFile);
+        formData.append("text", contentText);
+        PostsApi.createImagePost(formData).catch((err) => {
+            alert("FAILED");
+            console.error(err);
+        });
+        setContentFile(null);
+        setIsFilePicked(false);
+        window.location.reload();
+    };
 
-            <button className="button-post" onClick={handleSubmit}>
-              <img className="post" src={postURL} alt="Post" />
-              <span>Post</span>
-            </button>
-          </div>
+    const setFile = async (event) => {
+        setContentFile(event.target.files[0]);
+        setIsFilePicked(true);
+        console.log(`selected file is now: ${event.target.files[0]}`);
+    };
+
+    return (
+        <div className="postform-container">
+            <form className="postform">
+                <div>
```

Comment:
Looks like there are 2 pairs of div wrappers, could we pls remove one set?

Reply:
removed both because looked like they were not being used to style or hold things together.